src/Microsoft.Data.Sqlite/SqliteFactory.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Data.Common;
+
+namespace Microsoft.Data.Sqlite
+{
+    public class SqliteFactory : DbProviderFactory
+    {
+        private SqliteFactory()
+        {
+        }
+
+        private static SqliteFactory _instance;
+
+        public static SqliteFactory Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new SqliteFactory();
+                }
+                return _instance;
```

Comment:
Instead of making the property lazy, just set it in a static constructor. The code will be simpler, and initialization will still happen lazily the first time the type is used (i.e. the property is accessed)